### PR TITLE
docs(configurable-items): refactor configurable items stories - FE-3443

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour

This PR refactors ConfigurableItems stories to use `csf` / `mdx` formats instead of deprecated `storiesOf`

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Comment by DipperTheDan** [comment]
Not sure if this is a Storybook specific issue but this happens when I click and drag. Only on the Test story though. 
config-items-issue

File: src/components/configurable-items/configurable-items.stories.js [hunk 1]
```diff
@@ -38,73 +38,63 @@ const defaultConfigurableItemsData = [
...
-const store = new Store({
-  configurableItemsData: cloneDeep(defaultConfigurableItemsData),
-});
-const handleDrag = (draggedIndex, hoveredIndex) => {
-  const configurableItemsData = store.get("configurableItemsData");
-  const draggedItem = configurableItemsData[draggedIndex];
 
-  configurableItemsData[draggedIndex] = configurableItemsData[hoveredIndex];
-  configurableItemsData[hoveredIndex] = draggedItem;
+export const Default = () => {
```
Comment (DipperTheDan) [inline src/components/configurable-items/configurable-items.stories.js:42]: Left this comment regarding the drag and drop.
https://github.com/Sage/carbon/pull/3510#issuecomment-747532017

**Review by KatarzynaQA — APPROVED** [review]

File: src/components/configurable-items/configurable-items.stories.js (hunk 1 above)
Comment (mkrds) [reply src/components/configurable-items/configurable-items.stories.js:42]: Apparently one of the Storybooks story containers uses `transform: scale(1);` turning that off fixes the issue but I'm not sure if it's a good idea to do that though :| 
Zrzut ekranu 2020-12-18 o 15 50 56

**Review by DipperTheDan — DISMISSED** [review]

**Comment by nicktitchmarsh** [comment]
image

In the storybook preview the item seems to disappear when you grab it. Doesn't happen on canvas though.

**Review by nicktitchmarsh — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

File: src/components/configurable-items/configurable-items.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      const handleSave = (event) => {
+        event.persist();
+      };
+      const handleReset = () => {
+        setData(defaultConfigurableItemsData);
+      };
+      return (
+        <ConfigurableItems
+          onDrag={handleDrag}
+          onCancel={() => console.log("cancel")}
```
Comment (ianforster-sage) [inline src/components/configurable-items/configurable-items.stories.mdx:81]: Does this `console.log` pass linting without a disable line comment?

Comment (mkrds) [reply src/components/configurable-items/configurable-items.stories.mdx:81]: It's in MDX file, so ESlint does not work here :)

**Review by ianforster-sage — APPROVED** [review]

**Pull request merged**